clio-server/src/main/scala/org/broadinstitute/clio/server/dataaccess/elasticsearch/ElasticsearchDocumentMapper.scala

```diff
@@ -1,29 +1,52 @@
 package org.broadinstitute.clio.server.dataaccess.elasticsearch
 
+import io.circe.{Encoder, Json}
+import io.circe.syntax._
+import org.broadinstitute.clio.transfer.model.TransferMetadata
+import org.broadinstitute.clio.util.json.ModelAutoDerivation
+import org.broadinstitute.clio.util.model.UpsertId
+
 /**
   * Maps metadata to an Elasticsearch document.
   *
-  * @tparam ModelKey      The type of key.
-  * @tparam ModelMetadata The type of the metadata.
-  * @tparam Document      The type of the document.
+  * @tparam Key      The type of the TransferKey.
+  * @tparam Metadata The type of the TransferMetadata.
   */
-abstract class ElasticsearchDocumentMapper[ModelKey, ModelMetadata, Document] {
+class ElasticsearchDocumentMapper[Key <: Product: Encoder, Metadata <: TransferMetadata[
```

Comment:
We can probably remove the `<: TransferMetadata[Metadata]` type constraint. For what you're doing here, all that matters is that we've got an `Encoder` for it.